build.ts

```diff
@@ -54,25 +57,69 @@ export function loadTemplate(): Template[] {
 	})
 }
 
-export function renderTemplates(templates: Template[], data: Data): void {
-	templates.forEach((template) => {
-		const distPath = path.join(
-			path.resolve(config.distDir),
-			path.relative(config.rootDir, template.path).replace('pages', ''),
-			template.name + '.html'
+export function loadTemplatePartials(): Data {
+	const partialFiles = klawSync('./components', { nodir: true })
+	const partialData: Data = {}
+	partialFiles.forEach((file) => {
+		partialData[path.basename(file.path, '.mustache')] = fs.readFileSync(
+			file.path,
+			'utf8'
 		)
-		console.log(`Rendering ${distPath}`)
-		const output = Mustache.render(template.content, data)
-		fs.mkdirsSync(path.dirname(distPath))
-		fs.writeFileSync(distPath, output, 'utf8')
+	})
+	return partialData
+}
+
+export function renderTemplates(
+	templates: Template[],
+	globalData: Data,
+	partials: Data,
+	routes: Data
+): void {
+	templates.forEach((template) => {
+		if (template.name === '[pluginName]') {
+			console.log(`Rendering dynamic route pluginName`)
+			routes.pluginName.forEach((key: string) => {
+				const distPath = path.join(
+					path.resolve(config.distDir),
+					path.relative(config.rootDir, template.path).replace('pages', ''),
+					key,
+					'index.html'
+				)
+				console.log(`- Rendering ${distPath}`)
```

Comment:
When I run `yarn test`, I'm getting an error related to this `console.log`:
```
●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "- Rendering /tmp/tmp.nZkzTKEgzy/website-plugin-discovery/site/plugin/com.hieuthi.joplin.rubi-and-furigana/index.html".

       98 |             routes.pluginName.forEach((key) => {
       99 |                 const distPath = node_path_1.default.join(node_path_1.default.resolve(config.distDir), node_path_1.default.relative(config.rootDir, template.path).replace('pages', ''), key, 'index.html');
    > 100 |                 console.log(`- Rendering ${distPath}`);
          |                         ^
      101 |                 const data = Object.assign(Object.assign({}, globalData), { path: key, plugin: Object.assign({}, globalData.plugins.raw[key]) });
      102 |                 const output = mustache_1.default.render(template.content, data, partials);
      103 |                 fs_extra_1.default.mkdirsSync(node_path_1.default.dirname(distPath));

      at console.log (node_modules/@jest/console/build/BufferedConsole.js:187:10)
      at build.js:100:25
          at Array.forEach (<anonymous>)
      at build.js:98:31
          at Array.forEach (<anonymous>)
      at renderTemplates (build.js:95:15)
      at build.js:128:9
      at fulfilled (build.js:28:58)

```
Thank you for working on this!

Reply:
Nevermind! Applying the [suggested change below](https://github.com/joplin/website-plugin-discovery/pull/4#discussion_r973637157) fixes the issue for me!

Reply:
> Nevermind! Applying the [suggested change below](https://github.com/joplin/website-plugin-discovery/pull/4#discussion_r973637157) fixes the issue for me!

Thx for fixing this!

Reply:
That might not have fixed it — I'm sometimes running into the issue again (but not always...). Maybe the tests are timing out?

Reply:
it is something with the build script. No matter which function I call from the script, the following code will run and results something happening after the test is done.
```js
void (async function () {
	fs.ensureDirSync(config.distDir)
	clearBuildPath(config.distDir)
	copyStaticFiles(config.distDir)
	const template = loadTemplate()
	const globalData = await loadData()
	const partials = loadTemplatePartials()
	const routes = {
		pluginName: globalData.plugins.all.map((plugin: JoplinPlugin) => plugin.id),
	}
	renderTemplates(template, globalData, partials, routes)
})()

```

Reply:
Consider making that an `export`ed function. It doesn't need to be called in `build.test.ts`, but it can be called in `package.json` (see [this StackOverflow post about how to do that](https://stackoverflow.com/a/47962510)).

Reply:
Thx for pointing this out, I tried your method and it worked! It no longer runs the whole script every time I run some functions in the test which also eliminates the warnings. :)